web/src/components/njwds-extended/FileInput.tsx

```diff
@@ -0,0 +1,144 @@
+import { AcceptedFileType, InputFile } from "@businessnjgovnavigator/shared";
+import { ReactElement, useEffect, useLayoutEffect, useRef, useState } from "react";
+
+interface Props {
+  accept?: AcceptedFileType[];
+  className?: string;
+  errorMessage?: string;
+  hasError?: boolean;
+  helperText: string;
+  maxSizeInMegabytes?: number; // filesize in bytes (1,048,576 = 1 MB)
+  value: InputFile | undefined;
+  onChange: (file: InputFile) => void;
+}
+
+const FileTypeToInputString: Record<AcceptedFileType, string> = {
+  PDF: "application/pdf",
+  PNG: "image/png",
+};
+
+const InputStringToFileType: Record<string, AcceptedFileType> = (
+  Object.keys(FileTypeToInputString) as AcceptedFileType[]
+).reduce((acc, key) => {
+  const value = FileTypeToInputString[key];
+  acc[value] = key;
+  return acc;
+}, {} as Record<string, AcceptedFileType>);
+
+export const FileInput = (props: Props): ReactElement => {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const fileInput = require("../../../../node_modules/@uswds/uswds/packages/usa-file-input/src");
+  const fileInputRef = useRef<HTMLInputElement>(null);
+  const BYTES_IN_A_MB = 1048576;
+  const size = props.maxSizeInMegabytes ? props.maxSizeInMegabytes * BYTES_IN_A_MB : undefined;
+
+  const [hasFileBeenRead, setHasFileBeenRead] = useState<boolean>(false);
+  const [fileSizeError, setFileSizeError] = useState({ hasError: false, errorMessage: "" });
```

Comment:
It seems a little misleading to call this a `fileSizeError` when it also represents a file type error as well. Maybe just `fileUploadError` or something similar?

Reply:
included